### ISSUE TITLE
mat1 and mat2 shapes cannot be multiplied (522240x3 and 27x27)

### ISSUE THREAD

**Issue by JaegerWei123** [liyuantsao/FlowSR-LP#7]
Dear Author,
    I'm just a beginner, and when I execute the instruction _outputs = f.inear (inputs, self._weight, self.bias)_ from flow.py in LINF-LP, I get a RuntimeError: mat1 and mat2 shapes cannot be multiplied (522240x3 and 27x27). Does this also happen in your source code? The following is a detailed error report：

  File "G:\selflearningCV\test\BFSR-main\LINF-LP\test.py", line 301, in <module>
    res = eval_psnr(loader, model,
          ^^^^^^^^^^^^^^^^^^^^^^^^
  File "G:\selflearningCV\test\BFSR-main\LINF-LP\test.py", line 145, in eval_psnr
    z_lr_up = batched_predict_log_p(model, inp, coord, cell, lr_up_residual)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "G:\selflearningCV\test\BFSR-main\LINF-LP\test.py", line 43, in batched_predict_log_p
    _, z = model("query_log_p", inp=inp, feat=feat, coord=coord[:, row:row+256, :, :], cell=cell, gt=gt[:, :, row:row+256, :])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Anacoda\envs\Env1\Lib\site-packages\torch\nn\modules\module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Anacoda\envs\Env1\Lib\site-packages\torch\nn\modules\module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "G:\selflearningCV\test\BFSR-main\LINF-LP\models\linf.py", line 420, in forward
    return self.query_log_p(inp, feat, coord, cell, gt)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "G:\selflearningCV\test\BFSR-main\LINF-LP\models\linf.py", line 319, in query_log_p
    z, log_p = self.imnet(gt.permute(0, 2, 3, 1).contiguous().view(bs * w * h, -1), affine_info.permute(0, 2, 3, 1).contiguous().view(bs * w * h, -1))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Anacoda\envs\Env1\Lib\site-packages\torch\nn\modules\module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Anacoda\envs\Env1\Lib\site-packages\torch\nn\modules\module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "G:\selflearningCV\test\BFSR-main\LINF-LP\models\flow.py", line 47, in forward
    z, log_det_J = self.linears[i](z)
                   ^^^^^^^^^^^^^^^^^^
  File "D:\Anacoda\envs\Env1\Lib\site-packages\torch\nn\modules\module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Anacoda\envs\Env1\Lib\site-packages\torch\nn\modules\module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "G:\selflearningCV\test\BFSR-main\LINF-LP\models\flow.py", line 106, in forward
    outputs = F.linear(inputs, self._weight, self.bias)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: mat1 and mat2 shapes cannot be multiplied (522240x3 and 27x27)
python-BaseException

Process finished with exit code 1

**Comment by liyuantsao** [comment]
Hi @JaegerWei123,

Have you added the _--patch_ argument in your command?

**Comment by JaegerWei123** [comment]
> Hi @JaegerWei123,
> 
> Have you added the _--patch_ argument in your command?

Yeah I have added, just same as what introduced in README.MD, should the patch defined as "store_true" or just nothing?

![捕获](https://github.com/user-attachments/assets/83782abe-192d-4a97-855c-72005b61d20a)
![捕获](https://github.com/user-attachments/assets/120cd352-baa7-4053-b3c2-2a36e45b8c06)
](url)

**Comment by JaegerWei123** [comment]
Ahhh I found the reason, I should define patch as a value but not as what shown in README (define nothing), thx for your patience!!!
